### ISSUE TITLE
Count changefeeds

### ISSUE THREAD

**Issue by encryptio** [rethinkdb/rethinkdb#5799]
A user on slack mentioned that they wanted to watch the results of `.filter().count()` (which I later optimized to `.getAll(index=...).count()`).

Currently, `.count().changes()` is not supported; it throws:
```
e: Cannot convert NUMBER to SEQUENCE in:
r.db("db").table("table").count().changes()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by danielmewes** [comment]
This is a special case of https://github.com/rethinkdb/rethinkdb/issues/1118

**Comment by danielmewes** [comment]
On the current `v2.3.x`, this work-around based on the generic rewrite described in https://github.com/rethinkdb/rethinkdb/issues/1118#issuecomment-222234293 can be used.

Just set input_stream to whatever you want to have the count changefeed on, and set `II` and `IS` to `true` or `false` depending on whether you want to have initial results (`includeInitial`) and states (`includeStates`) respectively.

```js
var input_stream = r.table('t1');
var II = true;
var IS = true;

var f_BASE = 0;
var f_APPLY = function(acc, el) { return acc.add(1); };
var f_UNAPPLY = function(acc, el) { return acc.sub(1); };
var f_EMIT = function(acc) { return acc; };

input_stream.changes({includeInitial: true, includeStates: true}).fold(
  {f_acc: f_BASE, is_initialized: false},
  function(acc, el) {
    var f_acc = acc('f_acc');
    var new_f_acc = r.branch(el.hasFields("old_val"), f_UNAPPLY(f_acc, el('old_val')), f_acc).do(function(un_f_acc) {
        return r.branch(el.hasFields("new_val"), f_APPLY(un_f_acc, el('new_val')), un_f_acc);
      });
    var new_is_initialized = acc('is_initialized').or(el.hasFields('state').and(el('state').eq('ready')));
    return {f_acc: new_f_acc, is_initialized: new_is_initialized};
  },
  {emit: function(old_acc, el, new_acc) {
    var old_f_acc = old_acc('f_acc');
    var new_f_acc = new_acc('f_acc');
    var old_val = f_EMIT(old_f_acc);
    var new_val = f_EMIT(new_f_acc);
    // We handle the 'ready' state separately below
    var emit_state = r.expr(IS).and(el.hasFields('state')).and(r.expr(II).not().or(el('state').ne('ready')));
    var emit_update = old_acc('is_initialized').and(old_val.ne(new_val));
    var emit_initial = r.expr(II).and(old_acc('is_initialized').not().and(new_acc('is_initialized')));
    return r.branch(
      emit_state, [el],
      emit_update, [{'old_val': old_val, 'new_val': new_val}],
      emit_initial, r.branch(IS, [{'new_val': new_val}, {state: "ready"}], [{'new_val': new_val}]),
      []
    );
  }})
```

Note that this query does not yet work with the currently released version 2.3.2 because of https://github.com/rethinkdb/rethinkdb/issues/5800 .

**Comment by danielmewes** [comment]
This is going to ship with 2.4 as part of https://github.com/rethinkdb/rethinkdb/issues/1118 . Closing as duplicate since we're implementing the more generic functionality that includes this.
